packages/config/docs/main.md

```diff
@@ -0,0 +1,220 @@
+which is
+[passed as a flag to `@netlify/config`](https://github.com/netlify/build/blob/558fe8869f47ecc84d05bd4d26d32df00f47a3b9/packages/build/src/steps/update_config.js#L13)
+and has the highest merging priority.
+
+The logic in `@netlify/build` takes into account that the Netlify configuration can be mutated at any point. While most
+configuration properties can be mutated,
+[some are still read-only](https://github.com/netlify/build/blob/558fe8869f47ecc84d05bd4d26d32df00f47a3b9/packages/config/src/mutations/apply.js#L52)
+due to that additional complexity.
+
+In production, any configuration mutation
+[is temporarily persisted to `netlify.toml`](https://github.com/netlify/build/blob/558fe8869f47ecc84d05bd4d26d32df00f47a3b9/packages/config/src/mutations/update.js#L17)
+before deploying the site, and restored afterwards.
+
+### Netlify CLI
+
+Most Netlify CLI commands rely on configuration properties. The base command loads it
+[using `@netlify/config` programmatically](https://github.com/netlify/cli/blob/f87d9e6e1749bedaabaace3dae98ca1ed6d84fd6/src/commands/base-command.js#L474).
+
+An `inlineConfig` flag was added to `@netlify/config` so Netlify CLI can override specific configuration properties
+based on its own CLI flags, but this is not used yet (see https://github.com/netlify/cli/issues/1265,
+https://github.com/netlify/cli/issues/960, https://github.com/netlify/cli/issues/961).
+
+We try to
+[emulate the same environment variables](https://github.com/netlify/build/blob/558fe8869f47ecc84d05bd4d26d32df00f47a3b9/packages/config/src/env/main.js#L13)
+as on the buildbot, as much as possible.
+
+### Automated tests
+
+A few flags are available for programmatic users. Those are mostly used by our automated tests:
+
+- [The `testOpts` flag](https://github.com/netlify/build/blob/558fe8869f47ecc84d05bd4d26d32df00f47a3b9/packages/config/src/bin/flags.js#L153)
+- [The `buffer` flag](https://github.com/netlify/build/blob/558fe8869f47ecc84d05bd4d26d32df00f47a3b9/packages/config/src/bin/flags.js#L166)
+  which allows returning the output programmatically instead of printing it on the terminal
+
+## File paths
+
+The Netlify configuration relies on many file paths. Those are documented [here](file_paths.md).
+
+## UI build settings
+
+Users can specify the configuration either in files (`netlify.toml`, `_headers`, `_redirects`) or in the Netlify UI
+("build settings"). The latter has lower merging priority.
+
+UI build settings are passed to `@netlify/config` using
+[the `defaultConfig` flag](https://github.com/netlify/build/blob/558fe8869f47ecc84d05bd4d26d32df00f47a3b9/packages/config/src/default.js#L6).
+
+In the buildbot, this is populated
+[with some values](https://github.com/netlify/buildbot/blob/7383bb8971c718de9618a700ec5049cb71af7102/bot/buildbot.go#L135-L144)
+coming from the Bitballoon API.
+
+When called from the Netlify CLI, API calls to Bitballoon
```

Comment:
[sand] What do you think about using a more familiar term than `Bitballoon`?
We could phrase this as "these values are retrieved from our backend API" or something similar.
That should make sense to both internal and external contributors, WDYT?

Reply:
Fixed :+1: